### PR TITLE
fix: style and file transport bug - get output file path from user 

### PR DESCRIPTION
**Pull request by asafMasa**

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -3,111 +3,122 @@
...
-        throw new Error('The property host is omitted from config option log2httpServer. Please fill this field or remove the property.');
-    }
+  if (!httpOptions.host) {
+    console.warn('The property host is mandatory in config option log2httpServer. httpTransport won\'t be added.');
+    return false;
+  }
+  return true;
 }
 
 function generateFileTransport(level) {
```
Comment (cl-shaharshavit-rnd) [inline logger.js:17]: if you want different transports for different layer you should use level property

File: testFile.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const {  transports } = require('winston');
+const { transports } = require('winston');
```
Comment (cl-shaharshavit-rnd) [inline testFile.js:4]: if u want to actually test it you should reassign stdout and check that the logs are written to there

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: logger.js (hunk 1 above)
Comment (asafMasa) [reply logger.js:17]: that's an overkill usually all transports are open for the defined global log level

File: logger.js [hunk 3]
```diff
@@ -3,111 +3,122 @@
...
+    if (!loggerTransports.length) {
+      console.warn('No configuration was provided, adding default console logger.');
+      loggerTransports.push(new transports.Console());
     }
+    return loggerTransports;
+  }
+
+  getFileTransports() {
+    const path = require('path');
+    const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (asafMasa) [reply logger.js:108]: good point 👍

File: testFile.js (hunk 2 above)
Comment (asafMasa) [reply testFile.js:4]: good point 👍

File: logger.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (asafMasa) [reply logger.js:99]: fixed

File: testFile.js (hunk 2 above)
Comment (asafMasa) [reply testFile.js:4]: fixed

**Pull request merged**